src/compile/selection/interval.ts

```diff
@@ -1,104 +1,198 @@
-import {NewSignal, OnEvent, Stream} from 'vega';
+import {isObject, NewSignal, OnEvent, SignalValue, Stream} from 'vega';
 import {array, stringValue} from 'vega-util';
 import {SelectionCompiler, SelectionComponent, STORE, TUPLE, unitName} from '.';
-import {ScaleChannel, X, Y} from '../../channel';
+import {GeoPositionChannel, LATITUDE, LONGITUDE, ScaleChannel, X, Y} from '../../channel';
+import {FieldName} from '../../channeldef';
 import {warn} from '../../log';
 import {hasContinuousDomain} from '../../scale';
-import {SelectionInitInterval} from '../../selection';
-import {keys} from '../../util';
+import {IntervalSelectionConfigWithoutType, SelectionInitInterval, SELECTION_ID} from '../../selection';
+import {keys, vals} from '../../util';
+import {LayoutSizeIndex} from '../layoutsize/component';
+import {isUnitModel} from '../model';
 import {UnitModel} from '../unit';
 import {assembleInit} from './assemble';
 import {SelectionProjection, TUPLE_FIELDS} from './project';
 import scales from './scales';
 
 export const BRUSH = '_brush';
 export const SCALE_TRIGGER = '_scale_trigger';
+export const GEO_INIT_TICK = 'geo_interval_init_tick'; // Workaround for https://github.com/vega/vega/issues/3481
```

Comment:
Can you add something to the issue about how we need to update Vega-Lite when the issue is fixed in Vega?

Reply:
Done!